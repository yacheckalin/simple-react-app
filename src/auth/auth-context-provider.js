import React, { useState, useContext } from "react";
import { auth } from "./helpers";

const context = {
  isAuthorised: false,
};

export const AuthContext = React.createContext(context);
export const useAuthContext = () => useContext(AuthContext);

const errorResponse = JSON.stringify({
  result: "error",
  error:
    "\u041d\u0435\u0432\u0435\u0440\u043d\u044b\u0435 \u043f\u0430\u0440\u0430\u043c\u0435\u0442\u0440\u044b",
});
const successResponse = JSON.stringify({ result: "ok" });

const AuthContextProvider = ({ children }) => {
  const [isAuthorised, setIsAuthorised] = useState(true);
  const [loginError, setLoginError] = useState("");

  const login = ({ email: login, password }) => {
    //TODO: change for axios request
    let resp = auth({ email: login, password })
      ? successResponse
      : errorResponse;
    const response = Promise.resolve(resp);

    response.then((response) => {
      const res = JSON.parse(response);
      if (res.error) {
        setLoginError(res.error);
        setIsAuthorised(false);
      } else {
        setIsAuthorised(true);
      }
    });
  };

  const logout = ({ email, password }) => {
    //TODO: change for axios request
    const response = Promise.resolve(successResponse);

    response.then((resp) => {
      const res = JSON.parse(resp);

      if (!res.error) {
        setLoginError("");
        setIsAuthorised(false);
      }
    });

    setIsAuthorised(false);
    setLoginError("");
  };

  const resetLoginError = () => {
    setLoginError("");
  };

  return (
    <AuthContext.Provider
      value={{
        isAuthorised,
        setIsAuthorised,
        login,
        logout,
        loginError,
        resetLoginError,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

export default AuthContextProvider;
