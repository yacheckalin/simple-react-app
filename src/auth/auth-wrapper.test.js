import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

import Authorised from "./auth-wrapper";
import * as AuthContext from "./auth-context-provider";

describe("<Authorized />", () => {
  test("displays and renders properly", () => {
    const wrapper = shallow(<Authorised />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("display secured content", () => {
    const contextValues = { isAuthorised: true };
    jest
      .spyOn(AuthContext, "useAuthContext")
      .mockImplementation(() => contextValues);

    const TestElement = () => <div id="test-component">Some Test Element</div>;

    const wrapper = mount(
      <Authorised>
        <TestElement />
      </Authorised>
    );

    expect(wrapper.find("#test-component")).toBeTruthy();
  });

  test("hide secured content", () => {
    const contextValues = { isAuthorised: false };
    jest
      .spyOn(AuthContext, "useAuthContext")
      .mockImplementation(() => contextValues);

    const TestElement = () => <div id="test-component">Secured Content</div>;

    const wrapper = mount(
      <Authorised>
        <TestElement />
      </Authorised>
    );

    expect(wrapper.find("#test-component")).toHaveLength(0);
  });
});
