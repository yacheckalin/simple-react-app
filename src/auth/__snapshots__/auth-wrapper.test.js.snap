// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authorized /> displays and renders properly 1`] = `""`;
